tcms/telemetry/api/telemetry.py

```diff
@@ -0,0 +1,61 @@
+from django.db.models import Count
+from modernrpc.core import rpc_method
+
+from tcms.management.models import Priority
+from tcms.testcases.models import TestCase, Category, TestCaseStatus
+
+
+@rpc_method(name='Testing.breakdown')
+def breakdown(query=None):
+    """
+    .. function:: XML-RPC Testing.breakdown(query)
+
+        Perform a search and return the statistics for the selected test cases
+
+        :param query: Field lookups for :class:`tcms.testcases.models.TestCase`
+        :type query: dict
+        :return: Object, containing the statistics for the selected test cases
+        :rtype: dict
+    """
+
+    if query is None:
+        query = {}
+
+    test_cases = TestCase.objects.filter(**query)
+
+    count = {
+        'manual': test_cases.filter(is_automated=False).count(),
+        'automated': test_cases.filter(is_automated=True).count(),
+        'all': test_cases.count()
+    }
+
+    priorities_qs = Priority.objects.filter().annotate(
+        num_cases=Count('priority_case')
+    ).filter(
+        num_cases__gt=0,
+        priority_case__in=test_cases
+    )
+    priorities = [[priority.value, priority.num_cases] for priority in priorities_qs]
```

Comment:
This is totally wrong. For 67 Test Cases which I have in the DB that results in a P1 count of over 4000.
See the query below:
`TestCase.objects.all().values('priority').annotate(num_priority=Count('priority'))`

replace `all()` with `filter(**query)`.

Everything must be coming from TestCase wherever possible.